core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -2813,13 +2813,19 @@ public String toString() {
               OperatingSystemMXBean operatingSystemMXBean =
                   (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
               return operatingSystemMXBean.getTotalPhysicalMemorySize() * 2 / 3;
-            } catch (Exception e) {
+            } catch (Throwable e) {
               // The package com.sun.management may not be available on every platform.
               // fallback to a reasonable size.
               return "1GB";
             }
           }, "2/3 of total system memory, or 1GB if system memory size cannot be determined")
-          .setDescription("Memory capacity of each worker node.")
+          .setDescription("Memory capacity of each worker node. "
+                  + "It is recommended to set this value explicitly. "
+                  + "If you are running Alluxio workers in a containerized environment, "
+                  + "your Java runtime may return incorrect system memory "
+                  + "(i.e. the host memory instead of the container memory). "
+                  + "You should upgrade your JDK version according to "
+                  + "https://bugs.openjdk.java.net/browse/JDK-8226575.")
```

Comment:
I preferred not mention this line since the link isn't suggested the upgrade clearly and it's not guaranteed that the user will not see the issue in newer version (unless we test it)

Reply:
I think we should mention this pitfall, so that if anyone happens to meet this issue we can refer to the doc. I agree seeing is believing (and I have not tested different java versions). But testing this is non-trivial 0.0
@apc999 What do you think?

Reply:
@jiacheliu3 After some investigation which I mentioned in the issue, I will prefer to not mention the detailed reason here but only change the Exception to Throwable.
I doubt if the reason is true that the Java runtime return the host memory. From what I see, the JVM loads the correct jdk file [Loaded sun.management.OperatingSystemImpl from /usr/lib/jvm/java-1.8-openjdk/jre/lib/rt.jar] but the class somehow doesn't contain that method.

Reply:
I'm thinking this might also be related to classloading issues in your container environment.
even the jdk files are loaded, if by the wrong classloader, we may still the link error.

Reply:
agree, let's not mention JDK upgrade for now.
we can leave a TODO in src code

Reply:
@apc999 @LuQQiu Sorry about the confusion.

The comments here are related to the known bug in JDK when calling `operatingSystemMXBean.getTotalPhysicalMemorySize()`, not related to #12882. Using a higher version JDK will give you "correct" physical memory size, but NOT resolve the UnsatisfiedLinkError.

But it makes sense we discuss separately so I removed this part from the PR.